docs/reference/api-v3.yml

```diff
@@ -0,0 +1,166 @@
+openapi: 3.1.0
+x-stoplight:
+  id: 0x3135wk4xq0t
+info:
+  title: Open Food Facts Open API V3 - under development
+  description: |
+    As a developer, the Open Food Facts API allows you to get information 
+    and contribute to the products database. You can create great apps to 
+    help people make better food choices and also provide data to enhance the database.
+  termsOfService: 'https://openweathermap.org/terms'
+  contact:
+    name: Open Food Facts
+    url: 'https://slack-ssl-openfoodfacts.herokuapp.com/'
+    email: contact@openfoodfacts.org
+  license:
+    name: 'License (MIT, Apache 2.0, etc)'
+    url: 'https://opendatacommons.org/licenses/odbl/summary/index.html'
+  version: '3'
+servers:
+  - url: 'https://world.openfoodfacts.org'
+    description: prod
+  - description: dev
+    url: 'https://off:off@world.openfoodfacts.net'
+paths:
+  '/api/v3/product/{barcode}':
+    get:
+      tags:
+        - Read Requests
+      summary: Get information for a specific product by barcode (API V3 - Not yet implemented)
+      parameters:
+        - name: barcode
+          in: path
+          description: |
+            The barcode of the product to be fetched
+          required: true
+          style: simple
+          explode: false
+          schema:
+            type: string
+            example: '3017620422003'
+      responses: {}
+      description: Retrieve information for a product with a specific barcode.
+      operationId: get-product-by-barcode
+    parameters:
+      - schema:
+          type: string
+        name: barcode
+        in: path
+        required: true
+  /api/v3/product:
```

Comment:
Didn't you forgot the barcode parameter ? In fact I think the POST entry is just a new entry under the above path !

Reply:
```suggestion
```

Reply:
I was thinking about it, using the same path for both reading and writing, but that means we really have to use the method GET or POST to differentiate the two. It also means the complete "product id" must be in the URL, which is not only the barcode (e.g. on the producers platform, we also need an organization id).

So I'm not quite sure if it's such a good idea to have the same path for READ and WRITE.

Reply:
> we really have to use the method GET or POST to differentiate the two
Yes  that's really what REST advocates (and for caching and many scenario, this is far better)

I really think if we want to have a REST API, that it should be the same path, this is surprising for most developper not to do so.

Reply:
@stephanegigandet told he finally used the product id. So this should be reflected in this doc :-)